oeps/architectural-decisions/oep-0058-arch-translations-management.rst

```diff
@@ -0,0 +1,262 @@
+translators and reviewers across Transifex projects into the new project associated with
+the new repository. In addition, we can save all the progress the Open edX translators
+have accomplished by copying the Transifex Memory, the auto-translation feature that
+allows for Projects with similar strings to be automatically translated, from the old
+projects to this new one. Once older projects are made redundant by the new project, they
+will be deprecated. By moving all the translation files to the same repository we can
+increase the reach of the Transifex Memory feature to help translate similar strings
+across the entire code/documentation base.
+
+Get Translations Back for Deployment/Development
+================================================
+
+A new python library, called openedx-atlas, will be created. This will enable the
+placement of the translation files kept in openedx-translations into locally cloned
+repositories for development and containers containing the code translation files are
+formed from. This tool will manage the placement of translation files through an editable
+atlas configuration file (atlas.yml) kept in the repositories that have
+translation files kept in openedx-translations. The atlas.yml file will support
+options that allow for the concatenation, reorganization, and reformatting of translation
+files as they are copied to their locations amongst the code. The atlas.yml file
+will also support selecting which languages to be included in an Open edX deployment. The
+tool will have to be used/ran as part of the setup of a repository, whether for
+development or deployment.
+
+Impacts
+*******
+
+Impact on Translators
+=====================
+
+As we approach the end of the translation upgrade process, we will need to tactically
+move from multiple Transifex projects to a single project. This will require coordination
+with our translators to ensure that moving forward they are providing translations in the
+right place.
+
+Impact on Site Operators
+========================
+
+Currently the translation files for any given service or library is stored at the same
+place as the code, which has generally simplified the deployment story in the past. With
+this change, the translations files will move to their own repository. As we deprecate
+the old translations files, the relevant deployment tooling will need to be updated to
+pull down the translations from the new repository as a part of the deployment process.
+This will impact both the old Ansible based tooling as well as any new docker based
+tooling.
+
+Impact on Developers
+====================
+
+While it won’t directly impact the day-to-day workflow of developers, due to the same
+reasons that we impact site operators (new translations location), we will have to update
+development tools as well. In addition, we will create new instructions for developers on
+how to enable translations for a new service/repo when it comes online.
```

Comment:
> While it won’t directly impact the day-to-day workflow of developers

Isn't that the translation will no longer exists on their repo, thus to test or get translation at development time, developer would need to run `openedx-atlas` ?  So I assume that would **impact**  

Secondly, the degree of the impact which is not known yet... it seems every repo suppose to have the `atlas.yaml` file and install the new cli tool, I would argue if it would be possible for that to release itself in binary format, so 1)developer wouldn't have to think about their python version when they want to install it, which can be useful spefically in MFEs development..., i.e. _an MFE doesn't need to consider python env if the library has a binary release_ 2)Which can can make CI/CD workflow more effiecent as well.  

Note: _second comment might be releavnt in this scope I just wanted to raise because I think it can be helpful_

Reply:
How will developers receive errors from the source translation extract/compile step that may happen between openedx repos and openedx-translations? Currently, for example, if I misformat a translation's pluralization in edx-platform python, I can see [an error from django's compilemessages command](https://docs.djangoproject.com/en/4.1/topics/i18n/translation/#pluralization-var-notes), and CI ensures I have to repair it before the PR will merge.

Can we keep these compilation linting checks in place in CI even though the generated files are discarded from the repo? And can we encourage this practice on new repos too, including MFEs?

Reply:
@pomegranited 
I'm not familiar with this CI that checks the formatting of translations. Could you link it here? I agree that it should not be changed since its very useful. I also agree that this practice would benefit MFEs as well, though I'm less familiar with how that would work.

Reply:
@Carlos-Muniz  Ah.. looks like there aren't any CI checks for missing changes to source translations. I tested this by submitting https://github.com/openedx/edx-platform/pull/31276, to see whether CI would fail, and it didn't.

I could have sworn there were (using [`make detect_changed_source_translations`](https://github.com/openedx/edx-platform/blob/605a0a66e72343ef61c1a5310ed9acbbbf34915a/Makefile#L60-L61) or perhaps [`paver i18n*`](https://github.com/openedx/edx-platform/blob/master/pavelib/i18n.py)?), but I'm likely mis-remembering. Maybe the updated translation extractions are handled by the translations bot instead of CI?

Reply:
In some repositories we have that check as a part of the `quality` step